sdk/Pulumi/Deployment/Deployment_Inline.cs

```diff
@@ -80,7 +80,7 @@ internal static async Task<T> RunInlineAsyncWithResult<T>(IDeploymentBuilder bui
                 {
                     try
                     {
-                        var result = await runnerFunc(runner).ConfigureAwait(false);
+                        var result = await runner.RunAsync(()=> runnerFunc(runner)).ConfigureAwait(false);
```

Comment:
I don't think we actually need to pass the runner to runnerFunc here. It's not used in the provider implementation.